### ISSUE TITLE
Performance and Load Impact of 404 errors

### ISSUE THREAD

**Issue by mandolyte** [unfoldingWord-box3/book-package#9]
As an example, consider translation questions. In advance I do not know whether any questions exists for a given verse. So I have guess and attempt to fetch the questions. When they don't exist, I get one or two errors for each such failed attempt. I suspect it might be faster to request all the files in a repo and use it as a whitelist to avoid failed attempts.

```
Failed to load resource: the server responded with a status of 404 (Not Found)
:6060/#/BookPackage?id=bookpackagetn:1 

Access to XMLHttpRequest at 'https://git.door43.org/unfoldingWord/en_uhal/raw/branch/master/content/H0834a.md' from origin 'http://localhost:6060' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

**Comment by jag3773** [comment]
In the Gitea React Toolkit there is a method that allows you to get a tree view of a repositories contents.  I suspect that implementing that at a future point would be a good way to know before hand whether or not a file exists. For now performance it is not a priority.

**Comment by mandolyte** [comment]
I have discovered an easy way to solve this; Closing this issue and opening another to document what needs to be done.

**Comment by jag3773** [comment]
Which issue is the new one?

**Comment by mandolyte** [comment]
#27 is the new one
